go/libkb/active_device.go

```diff
@@ -298,3 +302,15 @@ func (a *ActiveDevice) SyncSecrets(m MetaContext) (err error) {
 	}
 	return RunSyncer(s, uid, true, nil)
 }
+
+func (a *ActiveDevice) CheckForUsername(m MetaContext, n NormalizedUsername) (err error) {
+	a.RLock()
+	uid := a.UID()
+	deviceID := a.DeviceID()
+	valid := a.valid()
+	a.RUnlock()
```

Comment:
Isn't there still a race right here?  ActiveDevice uid, deviceID can change while doing CheckDeviceForUIDAndUsername....

Reply:
Thought a little bit about the alternative.  Wrap all non-signup/non-login engines in a RWMutex.RLock(). Wrap signup/login in a Lock().  Need to be careful about reentrancy. Also probably want to fail the acquisition after a ~15s timeout or something.  It seemed pretty scary though...